flocker/control/test/test_persistence.py

```diff
@@ -57,56 +64,100 @@
-        return service
+        def updated(_):
+            self.assertEqual(
+                self.persistence_service.get(),
+                TEST_DEPLOYMENT.set(leases=update(original_leases)))
+        d.addCallback(updated)
+        return d
 
     def test_expired_lease_removed(self):
         """
         A lease that has expired is removed from the persisted
         configuration.
+        """
+        timestep = 100
+        node_id = uuid4()
+        ids = uuid4(), uuid4()
+        # First dataset lease expires at timestep:
+        now = self.clock.seconds()
+        leases = Leases().acquire(
+            datetime.fromtimestamp(now, UTC), ids[0], node_id, timestep)
+        # Second dataset lease expires at timestep * 2:
+        leases = leases.acquire(
+            datetime.fromtimestamp(now, UTC), ids[1], node_id, timestep * 2)
+        new_config = Deployment(leases=leases)
+        d = self.persistence_service.save(new_config)
+
+        def saved(_):
+            self.clock.advance(timestep - 1)  # 99
+            before_first_expire = self.persistence_service.get().leases
+            self.clock.advance(2)  # 101
+            after_first_expire = self.persistence_service.get().leases
+            self.clock.advance(timestep - 2)  # 199
+            before_second_expire = self.persistence_service.get().leases
+            self.clock.advance(2)  # 201
+            after_second_expire = self.persistence_service.get().leases
+
+            self.assertTupleEqual(
+                (before_first_expire, after_first_expire,
+                 before_second_expire, after_second_expire),
+                (leases, leases.remove(ids[0]), leases.remove(ids[0]),
+                 leases.remove(ids[0]).remove(ids[1])))
+        d.addCallback(saved)
+        return d
 
-        XXX Leases cannot be manipulated in this branch. See FLOC-2375.
-        This is a skeletal test that merely ensures the call to
-        ``update_leases`` takes place when ``_expire`` is called and should
-        be rewritten to test the updated configuration once the configuration
-        is aware of Leases.
+    @capture_logging(None)
+    def test_expire_lease_logging(self, logger):
         """
-        service = self.service()
-        d = service._expire()
+        An expired lease is logged.
+        """
+        node_id = uuid4()
+        dataset_id = uuid4()
+        leases = Leases().acquire(
+            datetime.fromtimestamp(self.clock.seconds(), UTC),
+            dataset_id, node_id, 1)
 
-        def check_expired(updated):
-            self.assertIsNone(updated)
+        d = self.persistence_service.save(Deployment(leases=leases))
 
-        d.addCallback(check_expired)
+        def saved(_):
+            logger.reset()
+            self.clock.advance(1000)
+            assertHasMessage(self, logger, _LOG_EXPIRE, {
+                u"dataset_id": dataset_id, u"node_id": node_id})
+        d.addCallback(saved)
         return d
 
-
```

Comment:
This change is causing the lint test to fail

http://build.clusterhq.com/builders/flocker-lint/builds/9364